aws-android-sdk-pinpoint/src/main/java/com/amazonaws/mobileconnectors/pinpoint/targeting/notification/EventSourceType.java

```diff
@@ -0,0 +1,194 @@
+import static com.amazonaws.mobileconnectors.pinpoint.targeting.notification.NotificationClientBase.CAMPAIGN_ID_ATTRIBUTE_KEY;
+import static com.amazonaws.mobileconnectors.pinpoint.targeting.notification.NotificationClientBase.CAMPAIGN_PUSH_KEY_PREFIX;
+import static com.amazonaws.mobileconnectors.pinpoint.targeting.notification.NotificationClientBase.JOURNEY_ACTIVITY_ID_ATTRIBUTE_KEY;
+import static com.amazonaws.mobileconnectors.pinpoint.targeting.notification.NotificationClientBase.JOURNEY_ID_ATTRIBUTE_KEY;
+
+public class EventSourceType {
+    private static final String TAG = EventSourceType.class.getSimpleName();
+    private static final String AWS_EVENT_TYPE_OPENED = "opened_notification";
+    private static final String AWS_EVENT_TYPE_RECEIVED_FOREGROUND = "received_foreground";
+    private static final String AWS_EVENT_TYPE_RECEIVED_BACKGROUND = "received_background";
+    private static final String PINPOINT_ATTRIBUTE_KEY = "pinpoint";
+    private static final String JOURNEY_ATTRIBUTE_KEY = "journey";
+    private final EventSourceAttributeParser attributeParser;
+    private String eventSourceName;
+    private String eventTypeOpenend;
+    private String eventTypeReceivedForeground;
+    private String eventTypeReceivedBackground;
+    private String eventSourceIdAttributeKey;
+    private String eventSourceActivityIdAttributeKey;
+    private String eventSourceKeyPrefix;
+
+    /**
+     * Campaign attributes are send from pinpoint flattened
+     * For example:
+     *  "pinpoint.campaign.campaign_id"
+     * Journey attributes come in JSON format
+     * This interface just seeks to abstract some of that away
+     * from the logic that handles the notfication
+     * @return
+     */
+    public interface EventSourceAttributeParser {
+        Map<String, String> getEventSourceAttributes(Bundle bundle);
+    }
+
+    private static final EventSourceAttributeParser CAMPAIGN_PARSER = new EventSourceAttributeParser() {
+        @Override
+        public Map<String, String> getEventSourceAttributes(Bundle bundle) {
+            if(bundle == null) {
+                return null;
+            }
+            final Map<String, String> campaignAttributes = new HashMap<>();
+            String campaignKeyPrefix = CAMPAIGN.getEventSourceKeyPrefix();
+
+            for(String key : bundle.keySet()) {
+                if (key.startsWith(campaignKeyPrefix)) {
+                    //If it matches the prefix, include it in the attributes
+                    //after stripping out the prefix
+                    campaignAttributes.put(key.replace(campaignKeyPrefix,""), bundle.getString(key));
+                }
+            }
+            return campaignAttributes.size() > 0 ? campaignAttributes : null;
+        }
+    };
+
+    private static final EventSourceAttributeParser JOURNEY_PARSER = new EventSourceAttributeParser() {
```

Comment:
Same here, consider splitting to `private static final class JourneyAttributeParser implements EventSourceAttributeParser`.

Reply:
See above